app/uk/gov/hmrc/exports/controllers/HeaderValidator.scala

```diff
@@ -17,22 +17,47 @@
 package uk.gov.hmrc.exports.controllers
 
 import javax.inject.Singleton
+import play.api.Logger
+import play.api.http.HeaderNames
 import uk.gov.hmrc.exports.models._
 @Singleton
 class HeaderValidator {
 
   def extractLrnHeader(headers: Map[String, String]): Option[String] = {
-    headers.get(CustomsHeaderNames.XLrnHeaderName)
+    extractHeader(CustomsHeaderNames.XLrnHeaderName, headers)
   }
 
   def extractDucrHeader(headers: Map[String, String]): Option[String] = {
-    headers.get(CustomsHeaderNames.XDucrHeaderName)
+    extractHeader(CustomsHeaderNames.XDucrHeaderName, headers)
   }
 
   def extractMrnHeader(headers: Map[String, String]): Option[String] = {
-    headers.get(CustomsHeaderNames.XMrnHeaderName)
+    extractHeader(CustomsHeaderNames.XMrnHeaderName, headers)
   }
 
+  def extractAuthTokenHeader(headers: Map[String, String]): Option[String] = {
+    extractHeader(HeaderNames.AUTHORIZATION, headers)
+  }
+
+  def extractConversationIdHeader(
+      headers: Map[String, String]): Option[String] = {
+    extractHeader(CustomsHeaderNames.XConversationIdName, headers)
+  }
+
+  def extractEoriHeader(headers: Map[String, String]): Option[String] = {
+    headers.get("X-EORI-Identifier")
+  }
+
+  private def extractHeader(headerName: String,
+                            headers: Map[String, String]): Option[String] = {
+    headers.get(headerName) match {
```

Comment:
I don't like this code. Why?
1) if option is defined return the same without any changes
2) if option is empty return None with Logger.

Here we need to only log a situation when the Option is not empty, we don't need to check anything else. I checked Option documentation and I found method that will be good for this:

https://www.scala-lang.org/api/current/scala/Option.html#orElse[B%3E:A](alternative:=%3EOption[B]):Option[B]

I checked documentation for version that we use and this method exists.

```
headers.get(headerName).orElse{
    Logger.error(s"Error Extracting $headerName")
    None
```

This solution is not probably the best one, but it's works and probably it should be faster:

```
/** Returns this $option if it is nonempty,
   *  otherwise return the result of evaluating `alternative`.
   *  @param alternative the alternative expression.
   */
  @inline final def orElse[B >: A](alternative: => Option[B]): Option[B] =
      if (isEmpty) alternative else this
```

I'm thinking there should be better solution, but I can't remember which one.

Reply:
I've added a guard so it will only return some when there is a value.